src/iop/hlrecovery.c

```diff
@@ -255,24 +282,228 @@ static inline int pos2plane(const int row, const int col, const uint32_t filters
   else return 1 + (row&1);
 }
 
+static void initial_gradients(const size_t w, const size_t height, float *restrict luminance, float *restrict distance, float *restrict gradient)
```

Comment:
all static routines must start with underscore, so should be `_initial_gradients`